src/index.ts

```diff
@@ -1,54 +1,85 @@
-import seq from 'seq-logging'
+import * as seq from 'seq-logging'
 import TransportStream from 'winston-transport'
 
+enum WinstonLevels {
+  error = 'error',
+  warn = 'warn',
+  info = 'info',
+  http = 'http',
+  verbose = 'verbose',
+  debug = 'debug',
+  silly = 'silly',
+}
+
 //
 // Inherit from `winston-transport` so you can take advantage
 // of the base functionality and `.exceptions.handle()`.
 //
-module.exports = class SeqTransport extends TransportStream {
-  private logger: seq.Logger
+class SeqTransport extends TransportStream {
+  public logger: seq.Logger
+
   constructor(
-    config: seq.SeqLoggerConfig, 
-    opts?: TransportStream.TransportStreamOptions
+    config: seq.SeqLoggerConfig,
+    opts: TransportStream.TransportStreamOptions = {}
   ) {
-    // options.handleExceptions 
+    // options.handleExceptions
     // - If true, info with { exception: true } will be written.
-    super(opts)
-    this.logger = new seq.Logger(config)    
+    super(Object.assign(opts))
 
-    setImmediate(() => this.emit('opened:seq'))
+    this.logger = new seq.Logger(config)
+    setImmediate(() => this.emit('opened'))
   }
 
   /**
  * Writes to the stream associated with this instance.
  * @param {Info} info - Winston log information.
  * @param {Function} next - Continuation to respond to when complete.
  * @returns {void}
- */  
+ */
   log(info: unknown, next: () => void): void {
+    const { level, message, ...rest } =
+    <{ level:string, message: string }>info
+
     setImmediate(() => {
-      // TODO: Perform the writing to seq instance
-      // Not sure whether to override the stream writer (TransportStream) 
-      //  or use the emit below.....
-
-      // this.logger.emit({
-      //   timestamp: new Date(),
-      //   level: 'Information',
-      //   messageTemplate: 'Hello for the {n}th time, {user}!',
-      //   // exception: ,
-      //   properties: {
-      //     user: process.env.USERNAME,
-      //     nn: 20,
-      //   },
-      // })
+      this.logger.emit({
+        timestamp: new Date(),
+        level: this.mapLevel(level),
+        messageTemplate: message,
+        // exception: ,
+        properties: rest,
+      })
+
+      setImmediate(() => this.emit('logged', info))
     })
 
-    this.emit('logged:seq', info)
     next()
   }
 
   close(): void {
-    this.logger.close()
-    setImmediate(() => this.emit('closed:seq'))
-  }  
+    setImmediate(() => {
+      this.logger.flush().then(() => {
+        setImmediate(() => this.emit('flushed', this.logger))
+        this.logger.close().then(() => {
+          setImmediate(() => this.emit('closed', this.logger))
+        })
+      })
+    })
+  }
+
+  private mapLevel (level: string) {
```

Comment:
There may not be any need for this anymore - recent Seq versions work reasonably with arbitrary level names, so it's preferable to pass these through. The `silly` and `http` levels will show as if they're `Information`, but that seems reasonable enough (and is more of a server-side problem than something to handle client-side, now).

Reply:
OK - I'll give it a try tonight and see how it goes

Reply:
Done :zap:

Reply:
😎 

Doesn't look like the changeset with these changes has been pushed yet?

Reply:
Sorry - this didn't work - reverted
About to push now